src/controller/controller.c

```diff
@@ -483,37 +483,62 @@ static bool controller_setup_node_connection_handler(Controller *controller) {
 
 static int controller_reset_heartbeat_timer(Controller *controller, sd_event_source **event_source);
 
+static int controller_check_node_liveness(Controller *controller, Node *node) {
+        uint64_t diff = 0;
+        uint64_t now = 0;
+
+        if (controller->heartbeat_threshold_msec <= 0) {
+                /* checking liveness of node by heartbeat disabled since configured threshold is <=0" */
+                return 0;
+        }
+
+        now = get_time_micros();
+        if (now == 0) {
+                bc_log_error("Failed to get the time");
+                return 0;
+        }
```

Comment:
I think we can pull this out of the `LIST_FOREACH` loop and calculate the current time only once. The `node->last_seen` timestamp won't change as long as we are in the heartbeat callback. Calculating it in the loop will also result in the `now` timestamp (very) slightly increasing - so calculating `now` once outside the loop is probably better. 
This was already there before, but I think we can improve here in the same go.

Reply:
done